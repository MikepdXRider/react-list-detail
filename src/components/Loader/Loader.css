.loader{
    width: 25%;
    min-width: 200px;
    height: 100px;

    color: white;
    font-size: 2rem;
    font-family: 'Rock Salt', cursive;
    display: flex;
    align-items: center;
    justify-content: center;

    background: rgba( 255, 255, 255, 0.1 );
    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
    backdrop-filter: blur( 10px );
    -webkit-backdrop-filter: blur( 10px );
    border-radius: 10px;
    border: 1px solid rgba( 255, 255, 255, 0.18 );
}

.loaderContainer{
    width: 100%;
    height: 100%;

    display: flex;
    justify-content: center;
    align-items: center;
}