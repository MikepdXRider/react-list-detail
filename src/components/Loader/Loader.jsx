import React from 'react'
import styles from './Loader.css'

export default function Loader() {
    return (
        <div className={styles.loaderContainer}>
            <p className={styles.loader}>Loading...</p>
        </div>
    )
}
