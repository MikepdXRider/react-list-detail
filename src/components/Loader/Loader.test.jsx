import { render, screen } from "@testing-library/react"
import Loader from './Loader.jsx'

it('renders loading component', () => {
    const {container} = render(
        <Loader />
    );

    screen.getByText(/Loading.../i);
    expect(container).toMatchSnapshot();
})