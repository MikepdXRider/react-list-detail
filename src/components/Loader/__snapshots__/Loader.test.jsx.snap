// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading component 1`] = `
<div>
  <article
    class="loaderContainer"
  >
    <p
      class="loader"
    >
      Loading...
    </p>
  </article>
</div>
`;
